code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -0,0 +1,465 @@
+	return ..()
+
+/obj/structure/displaycase/dvoraks_treat/trigger_alarm()
+	for(var/obj/structure/telecomns_doomsday_device/DD in range(7, get_turf(src)))
+		DD.thief = TRUE
+		break
+	return ..()
+
+/obj/item/remote_ai_upload // A 1 use AI upload. Potential D.V.O.R.A.K reward.
+	name = "remote AI upload"
+	desc = "A mobile AI upload. The bluespace relay will likely overload after one use. Make it count."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "dvorak_upload"
+	w_class = WEIGHT_CLASS_TINY
+	item_state = "camera_bug"
+	origin_tech = "syndicate=4;programming=6"
+	/// Integrated camera console to serve UI data
+	var/obj/machinery/computer/aiupload/dvorak/integrated_console
+
+/obj/machinery/computer/aiupload/dvorak
+	name = "internal ai upload"
+	desc = "How did this get here?! Please report this as a bug to github"
+	power_state = NO_POWER_USE
+	interact_offline = TRUE
+
+/obj/item/remote_ai_upload/Initialize(mapload)
+	. = ..()
+	integrated_console = new(src)
+
+/obj/item/remote_ai_upload/Destroy()
+	QDEL_NULL(integrated_console)
+	return ..()
+
+/obj/item/remote_ai_upload/attack_self(mob/user as mob)
+	integrated_console.attack_hand(user)
+
+/obj/item/remote_ai_upload/attackby(obj/item/O, mob/user, params)
+	if(istype(O, /obj/item/card/emag))
+		to_chat(user, "<span class='warning'>You are more likely to damage this with an emag, than achieve something useful.</span>")
+		return
+	var/time_to_die = integrated_console.attackby(O, user, params)
+	if(time_to_die)
+		to_chat(user, "<span class='danger'>[src]'s relay begins to overheat...</span>")
+		playsound(loc, 'sound/weapons/armbomb.ogg', 75, 1, -3)
+		sleep(5 SECONDS)
+		explosion(loc, -1, -1, 2, 4, flame_range = 4)
+		qdel(src)
+
+/obj/effect/spawner/lootdrop/telecomns_core_table
+	name = "telecomns core table spawner"
+	lootcount = 1
+	loot = list(
+			/obj/item/rcd/combat,
+			/obj/item/gun/medbeam,
+			/obj/item/mod/module/energy_shield,
+			/obj/item/storage/box/syndie_kit/oops_all_extraction_flares
+	)
+
+/obj/item/storage/box/syndie_kit/oops_all_extraction_flares
+	name = "surplus box of extraction flares"
+
+/obj/item/storage/box/syndie_kit/oops_all_extraction_flares/populate_contents()
+	for(var/I in 1 to 7)
+		new /obj/item/wormhole_jaunter/contractor(src)
+/obj/effect/spawner/random_spawners/telecomns_emp_loot
+	name = "telecomns emp loot"
+	result = list(
+		/obj/item/grenade/empgrenade = 8,
+		/obj/item/gun/energy/ionrifle/carbine = 1,
+		/obj/item/gun/energy/ionrifle = 1)
+
+/obj/effect/spawner/random_spawners/telecomns_teleprod_maybe
+	name = "teleprod maybe"
+	result = list(
+		/datum/nothing = 4,
+		/obj/item/melee/baton/cattleprod/teleprod = 1)
+
+// This could work in any ruin. However for now, as the scope is quite large, it's going to be coded a bit more to D.V.O.R.A.K
+/obj/structure/environmental_storytelling_holopad
```

Comment:
Tiny dumb thing but should the holopads deactivate once DVORAK is destroyed?

Reply:
mmm, probably a good idea, or make it glitch out. Will look into that